src/bdk_cli.rs

```diff
@@ -117,8 +119,32 @@ where
         timeout: wallet_opts.electrum_opts.timeout,
     });
 
-    // Fall back to Electrum config if Esplora config isn't provided
-    let config = config_esplora.unwrap_or(config_electrum);
+    #[cfg(feature = "compact_filters")]
+    let compact_filter_config: Option<AnyBlockchainConfig> = {
+        let peerconfig = BitcoinPeerConfig {
+            address: wallet_opts.compactfilter_opts.address.clone(),
+            socks5: None,
```

Comment:
Noted. Also adding SOCKS5 optional credentials. 

Because of so many options, flag reuse clashes are happening. So had to remove `short` for some cases. Or we can use some other alphabets, but might not make intuitive sense.

Reply:
Proxy server configuration is extracted away into a separate `ProxyOpt` structure.